### ISSUE TITLE
Enable password protecting the webui on coordinators

### ISSUE THREAD

**Issue by nervetattoo** [arangodb/arangodb-mesos-framework#10]
It seems that currently the webui will be openly accessible from a coordinator-node if you can access this node. Having the option to either stop it listening on the coordinator endpoint, or better, password protect it, would be nice.

**Comment by m0ppers** [comment]
Yes this is a known issue. Unfortunately right now the DC/OS reverse proxy will remove all our Authentication headers to the coordinator because they are using these for their authentication stuff. We need to figure out with mesosphere how to fix that and this is the reason that there is currently no UI option when starting the coordinator.

You can still enable authentication manually using the args parameters (specify --server.authentication true and --server.jwt-secret) but then you will not be able to access the webinterface through DC/OS anymore and you will have to dig a tunnel or something like that to make that work.

**Comment by nervetattoo** [comment]
Ah. I see. My problem is not with the `dcos.cluster/service/arangodb3`, but rather the `ip:port` directly to the coordinator task where that is an exposed port from the container. So, given what you're saying, if I set up authentication args then I will not be able to use the reverse proxied `services/<service>` url, but I should be able to use the directly exposed coordinator port. Sounds as a fair enough fix though.

**Comment by nervetattoo** [comment]
This is unclear: Are all arguments passed to the `framework` command passed-through to all arangod instances?

**Comment by m0ppers** [comment]
Yes. This is purely a DC/OS UI issue....technically authentication is working within DC/OS...

The framework has its own set of arguments which controls how the individual tasks in DC/OS are being started.

These arguments here can be used to immediately control which (additional) arguments are being passed to the tasks:

https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/A/arangodb3/5/marathon.json.mustache#L50

**Comment by m0ppers** [comment]
These options are btw also visible in the UI when installing ArangoDB via the Universe :)

**Comment by nervetattoo** [comment]
Great! So, to be crystal clear, something like this should suffice:
```
--arangodb_additional_agent_args="--server.authentication true --server.jwt-secret = \"abc123\""
--arangodb_additional_dbserver_args="--server.authentication true --server.jwt-secret = \"abc123\""
--arangodb_additional_coordinator_args="--server.authentication true --server.jwt-secret = \"abc123\""
```
Is this the correct way to pass multiple additional args? It seems all instances runs the webui, so I assume all task types require the args.
Thanks!

**Comment by m0ppers** [comment]
yes...it is a simple string being passed to the docker container :)

**Comment by nervetattoo** [comment]
Thanks a bunch. One more follow-up question:
It's unclear how the framework works in this case. Re-configuring the service restarts the framework, but no tasks are restarted, meaning the configuration changes are not passed on.
Will I actually have to kill them one-by-one and wait for the framework to respawn new tasks with new configuration, or is there are better way to propagate the changes?

**Comment by m0ppers** [comment]
There is actually a possiblity to trigger that from the framework. It is not yet official but SHOULD work.

Try to do a POST to `<framework-url>/v1/restart.json `

**Comment by nervetattoo** [comment]
So, let me summarise my experience @m0ppers 
I set the extra args like this in DC/OS by editing the service in the UI and changing the JSON
```
"arangodb_additional_agent_args=\"--server.authentication true --server.jwt-secret abc123\""
"arangodb_additional_dbserver_args=\"--server.authentication true --server.jwt-secret abc123\"
"arangodb_additional_coordinator_args=\"--server.authentication true --server.jwt-secret abc123\"
```

Then i did a POST to `restart.json`
The framework then restarted, and began restarting the nodes. Now the agents failed to come up because of "unknown arguments". This basically broke the cluster because all agents failed to come back up, and when I redeployed with removing the args, it was unable to self-heal with no agency in place. 

I tried manually getting the agency back by starting the docker containers again, but the framework had moved to a new node, so while the agency managed to work fine between itself, it was no longer part of the framework and nothing self-healed. I ended up wiping arangodb3 and restoring from backup because I realised this would cause downtime in some way. I theorised that a fresh deploy with the extra args for just the coordinator would work ...

Redeploying with the extra-args in an `options.json` worked the same way; It didnt. Framework started, but all tasks failed due to unknown args.

It would be great if you guys could document a working example of _how to pass extra args_, and maybe even this exact use case. I'm happy to test and verify it further, but the next time I'm going to spin up a new DC/OS cluster for testing, and I need to figure out when I have time for that.

The good part is that the db itself continued working fine as I was blowing up the framework and the agency. Of course, when the db/coords tried to restart with the extra args, it crashed and burned.

**Comment by m0ppers** [comment]
sorry to hear that :( I retested and indeed it seems to have broken at some point. The options are the correct ones and should be understood by all roles. Maybe some escaping changed from 1.8 to 1.9.

**Comment by chanakyan-arivumani** [comment]
I'm stuck with the same issue :( 
Is there any workaround for this?

**Comment by dothebart** [comment]
Hi, we already discussed this issue with @joerg84 when this issue was opened. Maybe you could open an issue upstream and reference this issue in your report?
